src/github.com/t3chguy/riot-static/mxclient/utils.go

```diff
@@ -1,50 +0,0 @@
-// Copyright 2017 Michael Telatynski <7t3chguy@gmail.com>
-//
-// Licensed under the Apache License, Version 2.0 (the "License");
-// you may not use this file except in compliance with the License.
-// You may obtain a copy of the License at
-//
-//     http://www.apache.org/licenses/LICENSE-2.0
-//
-// Unless required by applicable law or agreed to in writing, software
-// distributed under the License is distributed on an "AS IS" BASIS,
-// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
-// See the License for the specific language governing permissions and
-// limitations under the License.
-
-package mxclient
-
-import "github.com/matrix-org/gomatrix"
-
-// Keeping here in case it becomes used again.
-//func ConcatEventsSlices(slices ...[]gomatrix.Event) []gomatrix.Event {
-//	var totalLen int
-//	for _, s := range slices {
-//		totalLen += len(s)
-//	}
-//	tmp := make([]gomatrix.Event, totalLen)
-//	var i int
-//	for _, s := range slices {
-//		i += copy(tmp[i:], s)
-//	}
-//	return tmp
-//}
-
-func ReverseEventsCopy(events []gomatrix.Event) []gomatrix.Event {
-	var newEvents []gomatrix.Event
-	for i := len(events) - 1; i >= 0; i-- {
-		newEvents = append(newEvents, events[i])
-	}
-	return newEvents
-}
-
-type StringIntPair struct {
-	Key   string
-	Value int
-}
-
-type StringIntPairList []StringIntPair
```

Comment:
You might be able to guess why I'm commenting on this. Take a moment to consider what I don't like about this type and its usage. Ready?

- The name is obvious. It's a slice of `StringIntPair` which, strangely enough, is a 2-field struct containing a string and an int. The name adds nothing useful.
- Prematurely made generic. Have you looked at actually where and how you use this? Maybe you haven't, but I sure have. You use this type to mean one thing and one thing only currently: A map of domains to user counts in a room. Why did you decide to make this of all things generic? I had to *decipher* the usage of this type based on [this line with no comments](https://github.com/t3chguy/riot-static/pull/6/files#diff-e7ec0d8b675b329896b0e8d7ee655e67L161). If you had just used a specific type name (e.g `DomainUserCounts`) and not tried to make this generic then this would've been infinitely easier and clearer for me to read.
- Missing explanation on the actual interesting part of this type: the `sort.Interface` implementation.